packages/flutter_tools/test/web.shard/hot_reload_web_errors_test.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@Tags(<String>['flutter-test-driver'])
+library;
+
+import 'package:file/file.dart';
+
+import '../integration.shard/test_data/hot_reload_const_project.dart';
+import '../integration.shard/test_driver.dart';
+import '../integration.shard/test_utils.dart';
+import '../src/common.dart';
+
+void main() {
+  late Directory tempDir;
+  final HotReloadConstProject project = HotReloadConstProject();
+  late FlutterRunTestDriver flutter;
+
+  final List<String> additionalCommandArgs = <String>[
+    '--extra-front-end-options=--dartdevc-canary,--dartdevc-module-format=ddc',
+  ];
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('hot_reload_test.');
+    await project.setUpIn(tempDir);
+    flutter = FlutterRunTestDriver(tempDir);
+  });
+
+  tearDown(() async {
+    await flutter.stop();
+    tryToDelete(tempDir);
+  });
+
+  testWithoutContext(
+    'hot reload displays an error message when removing a field from a const class',
+    () async {
+      await flutter.run(chrome: true, additionalCommandArgs: additionalCommandArgs);
+      project.removeFieldFromConstClass();
+
+      expect(
+        flutter.hotReload(),
+        throwsA(
+          isA<Exception>().having(
+            (Exception e) => e.toString(),
+            'message',
+            // TODO(srujzs): Change this to "Try performing a hot restart instead." once we emit
+            // that string in the delta inspector. Then unify this test with
+            // hot_reload_errors_test.dart.
+            contains('Const class cannot become non-const'),
```

Comment:
I'm guessing this is the code actually being used for the hot reload. 
https://github.com/flutter/flutter/blob/df114fbe9b30afc94ee1154393aef8b575ee1612/packages/flutter_tools/test/integration.shard/test_data/hot_reload_const_project.dart#L49

If this really is the error text you are getting it seems like we have a but in the delta inspector since the const keyword should still exist on the constructor.

Reply:
No, you're correct here, I should and do see a different error.

I have Flutter tools running with a local SDK (but not the test framework), so I copied over the text I saw in the console when hot reloading, but I was changing the constructor. I updated it to reflect the correct log.

Reply:
FYI - I also unified the tests here with the VM and pass the different error message as a parameter.